content/blog/2024/08/kubewarden-1-16-release.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Kubewarden 1.16 release
+authors:
+  - Víctor Cuadrado Juan
+date: 2024-08-19
+---
+
+# Kubewarden v1.16.0 release
+
+We are thrilled to announce the release of Kubewarden v1.16.0! Following the
+northern hemisphere summer, this version packs some goodies but is a bit more
+lightweight than usual.
+
+## kwctl scaffold for AdmissionRequests
+
+The `kwctl` cli has learned a new command, `kwctl scaffold admission-request`,
+which prints a Kubernetes AdmissionRequest object from the provided Kubernetes
+resource definition.
+
+This is useful when developing policies (and not only limited to Kubewarden ones). Now,
+you can scaffold the desired AdmissionRequest the policy would receive, and
+then end-to-end test a Kubewarden policy against the created AdmissionRequest with
+`kwctl run`. For example, here we scaffold a request to create an Ingress object,
+which then we use to test one of our policies out of cluster:
+
+```console
+$ kwctl scaffold admission-request --operation CREATE \
+  --object ingress.yaml > request-ingress.json
+
+2024-08-16T10:32:18.090011Z  INFO kwctl::scaffold::admission_request: Resource catalog not found, building it
+
+$ kwctl run \
+  --settings-json '{"constrained_labels": {"owner": ".*"}}' \
+  -r request-ingress.json \
+  registry://ghcr.io/kubewarden/policies/safe-labels:v0.1.5
+```
+
+This new command `kwctl scaffold admission-request` needs to connect to a
+Kubernetes APIserver at least for the first time, as some information cannot be
+inferred from the resource itself, such as:
+
+- Knowing if the resource is namespaced or not.
+- Obtaining the plural name of the resource (e.g: `pods`).
+
+Kwctl caches this information locally for subsequent runs. If the cluster is
+still available, it will use it to refresh the cache.
+
+Given that `kwctl` connects to the cluster, it also means that the new command
+can scaffold for CRDs deployed in that cluster.
+
+Currently, only the CREATE operation is supported, UPDATE and DELETE will be
+added in the future.
+
+## Policy updates
+
+Through this cycle, we have updated the following policies.
+
+### Container-resources policy
+
+This policy checks for resource limits on the containers, and mutates them to
+add limits if configured so.
+
+With its new release `v0.3.0`, the policy now will not mutate the resource
+with a limit that is less than the requested, but will reject the resource
+instead, to force the user to change the minimum request resource amount or
+adjust the policy configuration.
+
+### User-group-psp policy
+
+Starting with `v0.6.2`, if the container image validation is enabled (with
+`validate_container_image_configuration`), the policy will only check for it
+when the rule is "MustRunAs" or "MayRunAs"; "RunAsAny" does not check the
+container image.
```

Comment:
Actually, the container image is not enabled by default. In the `v0.6.2` we fix an issue in the `RunAsAny` rule for the group when the container image validation is enabled. Furthermore, we documented the issue about the timeout due the time that context aware call takes.